src/cable/handler.cr

```diff
@@ -17,58 +17,63 @@ module Cable
 
       ws = HTTP::WebSocketHandler.new do |socket, ws_ctx|
         connection = T.new(ws_ctx.request, socket)
-        connection_id = connection.connection_identifier
 
-        # we should not add any connections which have been rejected
-        Cable.server.add_connection(connection) unless connection.connection_rejected?
+        if connection.open? && connection.connection_accepted?
+          connection_id = connection.connection_identifier
```

Comment:
If the connection hasn't closed (i.e. the socket hasn't closed), and the connection hasn't been rejected, then go ahead and setup the socket handling. Otherwise it's pointless to setup, and we can just exit out of this block.

Reply:
Instead of putting everything into an `if` statement, you could move the exit check into a one-liner:

```suggestion
        # we should not add any connections which have been rejected
        next if connection.connection_rejected?
        Cable.server.add_connection(connection)
        connection_id = connection.connection_identifier
```

Reply:
I like that. I still need to check if the connection is closed, but I'll throw both of those in to the one-liner